sdk/core/core/src/az_span_builder.c

```diff
@@ -22,3 +22,56 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
   self->size += 1;
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_span_builder_replace(
+    az_span_builder * const self,
+    size_t start,
+    size_t end,
+    az_span const span) {
+  AZ_CONTRACT_ARG_NOT_NULL(self);
+  AZ_CONTRACT_ARG_NOT_NULL(&start);
```

Comment:
How could &start and &end EVER realistically be null?
I don't think we need these tests. start MUST be within self is a valid test.